py/dml/ctree.py

```diff
@@ -356,6 +358,28 @@ def toc(self):
 
 mkLabel = Label
 
+class UnrolledLoop(Statement):
+    @auto_init
+    def __init__(self, site, substatements, break_id):
+        assert isinstance(substatements, list)
+
+    def toc(self):
+        out('{\n', postindent = 1)
+        self.toc_inline()
+        out('}\n', preindent = -1)
+
+    def toc_inline(self):
+        for substatement in self.substatements:
+            substatement.toc()
+        if self.break_id is not None:
+            out(f'_unrolledbreak{self.break_id}: UNUSED;\n')
+
+    def control_flow(self):
+        bodyflow = mkCompound(self.site, self.substatements).control_flow()
+        return bodyflow.replace(fallthrough=True, br=False)
```

Comment:
how do we know that `fallthrough=True`? Consider e.g.: 
```
method m() -> (int) {
  #foreach x in ([0]) {
    return x;
  }
}
```
Wouldn't a more correct value be `fallthrough=bodyflow.fallthrough or bodyflow.br`?

Reply:
Oops. You're quite right.